Dev

Stellareign opened the pull request:
Обновлённая dev-ДЗ1

Stellareign committed:
Пересобранный проект

Stellareign committed:
Пересобранный проект

Stellareign committed:
Пересобранный проект

Stellareign committed:
Пересобранный проект

Stellareign committed:
переместила интефейсы в пакет
добавмила репозиторий юзера
добавила сущность картинки
Создала поля поля сущности пароля (он нужен отдельно???)

Stellareign committed:
переместила интефейсы в пакет
добавила репозиторий юзера
добавила сущность картинки
Создала поля поля сущности пароля (он нужен отдельно???)
Перенесла маппер в Application.clss
создала неполноценный маппер на юзера (рабочий ???)

Stellareign committed:
переместила интефейсы в пакет
добавила репозиторий юзера
добавила сущность картинки
Создала поля поля сущности пароля (он нужен отдельно???)
Перенесла маппер в Application.clss
создала неполноценный маппер на юзера (рабочий ???)

alkor63 committed:
Дописал все методы в контроллере+сервисе объявлений

alkor63 committed:
Дописал все методы в контроллере+сервисе объявлений

alkor63 committed:
Дописал все методы в контроллере+сервисе объявлений

alkor63 committed:
Дописал все методы в контроллере+сервисе объявлений

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

Stellareign committed:
Merge remote-tracking branch 'origin/RuAna' into RuAna

Stellareign committed:
переместила интефейсы в пакет
добавила репозиторий юзера
добавила сущность картинки
Создала поля поля сущности пароля (он нужен отдельно???)
Перенесла маппер в Application.clss
создала неполноценный маппер на юзера (рабочий ???)

Stellareign committed:
БД починились
Пробная регистрация в сваггере прошла

alkor63 committed:
повторная попытка пуша в dev

alkor63 committed:
повторная попытка пуша в dev

alkor63 committed:
Merge remote-tracking branch 'origin/Aleksey' into Aleksey

# Conflicts:
#	src/main/java/ru/skypro/homework/entity/Ad.java
#	src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java
#	src/main/resources/application.properties

Rumonz1 committed:
создал несколько классов по комментариям

Rumonz1 committed:
создал несколько классов по комментариям

Rumonz1 committed:
Подправил названия

Rumonz1 committed:
Подправил названия

Stellareign committed:
БД снова запускаются
контроллеры не работают :(

alkor63 committed:
Внес исправления в маппер, контроллер-сервер объявлений
Пока не разобрался почему в БД объявления корректно добавляются, редактируются и удаляются, но в Сваггере "код ошибки 500"

Rumonz1 committed:
фиксирую малые изменения, пока без логики

Rumonz1 committed:
построил логику контроллеров(не до конца)

Stellareign committed:
Удалила повторяющиеся классы и переместила интерфейс комментов в пакет

Stellareign committed:
Merge branch 'RuAna' into dev2

# Conflicts:
#	pom.xml
#	src/main/java/ru/skypro/homework/config/MapperUtil.java
#	src/main/java/ru/skypro/homework/config/WebSecurityConfig.java
#	src/main/java/ru/skypro/homework/controller/AdsController.java
#	src/main/java/ru/skypro/homework/controller/CommentsController.java
#	src/main/java/ru/skypro/homework/controller/UserController.java
#	src/main/java/ru/skypro/homework/dto/ads/AdsDTO.java
#	src/main/java/ru/skypro/homework/dto/ads/CreateOrUpdateAdDTO.java
#	src/main/java/ru/skypro/homework/dto/ads/ExtendedAdDTO.java
#	src/main/java/ru/skypro/homework/dto/comments/CommentsDTO.java
#	src/main/java/ru/skypro/homework/entity/Ad.java
#	src/main/java/ru/skypro/homework/entity/Comment.java
#	src/main/java/ru/skypro/homework/entity/Password.java
#	src/main/java/ru/skypro/homework/entity/User.java
#	src/main/java/ru/skypro/homework/repository/UserRepository.java
#	src/main/java/ru/skypro/homework/service/CommentsService.java
#	src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java
#	src/main/java/ru/skypro/homework/service/impl/CommentsServiceImpl.java
#	src/main/java/ru/skypro/homework/service/interfaces/AdsService.java

Stellareign committed:
Удалила повторяющиеся классы и переместила интерфейс комментов в пакет

Stellareign committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/entity/Ad.java
#	src/main/java/ru/skypro/homework/service/CommentsService.java

Stellareign committed:
Удалила повторяющиеся классы и переместила интерфейс комментов в пакет

alkor63 committed:
Нашел подсказки как победить зацикливание ссылок
Добавил пару зависимостей в pom.xml и создал class JacksonConfig

alkor63 committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/entity/User.java
#	src/main/resources/application.properties

Stellareign committed:
Несколько правок по юзерам

Stellareign committed:
Подправила добавление коммента

Stellareign committed:
Подправила добавление коммента

Stellareign committed:
Пользователь сохраняется в БД.

dmitry-bizin commented on pom.xml:
```diff
@@ -57,11 +58,49 @@
+    <dependency>
+      <groupId>com.fasterxml.jackson.datatype</groupId>
+      <artifactId>jackson-datatype-jsr310</artifactId>
+      <version>2.13.2</version>
```
версию писать не нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/JacksonConfig.java:
```diff
@@ -0,0 +1,23 @@
+@Configuration
+public class JacksonConfig {
+    @Bean
+    public Jackson2ObjectMapperBuilderCustomizer addCustomBigDecimalDeserialization() {
```
а зачем так кастомизировать? вы просто замалчиваете ошибку, что у вас поля называются в DTO не так, как должны, код в итоге всё равно работать не будет. Класс этот надо удалить

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,152 @@
+
+    @PostMapping
+    @Operation(summary = "Добавление нового объявления")
+    public ResponseEntity<Ad> createAd(@RequestBody String title,   // 'заголовок объявления'
+                                       int price,               // 'цена объявления'
+                                       String description,            //'описание объявления'
+                                       String imagePath)             //'адрес картинки объявления'
```
```suggestion
    public ResponseEntity<Ad> createAd(@RequestPart("properties") CreateOrUpdateAdDTO createOrUpdateAdDTO, @RequestPart("image") MultipartFile image)
```
вот так должно быть

dmitry-bizin replied:
по-прежнему актуально

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,152 @@
+        return ResponseEntity.ok().body(new AdsDTO(adList.size(), adList));
+    }
+
+    @PostMapping
```
```suggestion
    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
```
вот так должно быть

dmitry-bizin replied:
по-прежнему актуально

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,152 @@
+                                       String imagePath)             //'адрес картинки объявления'
+        //                                      User author) {            //'id автора объявления'
+         {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String currentPrincipalName = authentication.getName(); // имя того, кто давит на клавиши
+        User currentUser = mapperUtil.getMapper().map(authentication, User.class);
+        if (currentUser == null) return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        currentUser = userRepository.getById(1); // временно, пока не разберусь с авторизованным юзером
+            CreateOrUpdateAdDTO createAdDTO = new CreateOrUpdateAdDTO(title, price, description);
+            Ad newAd = adsService.addAd(mapperUtil.createAdfromDTO(createAdDTO,imagePath, currentUser));
+            if (newAd != null) {
+//                return ResponseEntity.status(HttpStatus.CREATED).body(newAd);
+                return ResponseEntity.ok().body(newAd);
+            }
+          else {
+                return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+            }
```
никакой логики в контроллерах быть не должно - всё должно быть в сервисах

dmitry-bizin replied:
по-прежнему актуально

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,152 @@
+    @GetMapping("/{adId}")
+    @Operation(summary = "Получение информации об объявлении по id")
+    public ResponseEntity<ExtendedAdDTO> getAdById(@PathVariable int adId) {
+        Optional<Ad> adById = adsService.getAdById(adId);
+        if (adById.isEmpty()) return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+//        User authorAd = userRepository.findById(adId);
+//      if (authorAd == null) return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        ExtendedAdDTO extendedAdDTO = mapperUtil.createExtendedAdDTO(adById.get());
+        return ResponseEntity.ok().body(extendedAdDTO);
```
никакой логики в контроллерах быть не должно - всё должно быть в сервисах

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,152 @@
+//      "title": "string",
+//              "price": 10000000,
+//              "description": "string"
+    public ResponseEntity<Ad> updateAd(@PathVariable int adId, @RequestParam String title, int price, String description) {
```
```suggestion
    public ResponseEntity<Ad> updateAd(@PathVariable int adId, @RequestBody CreateOrUpdateAdDTO createOrUpdateAdDTO) {
```
вот так должно быть

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,152 @@
+
+    @PatchMapping("/{adId}/image")
+    @Operation(summary = "Обновление картинки объявления")
+    public ResponseEntity<Ad> editImageAdById(@PathVariable int adId, @RequestParam String imagePath) {
```
```suggestion
    public ResponseEntity<Ad> editImageAdById(@PathVariable int adId, @RequestParam("image") MultipartFile image) {
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,152 @@
+        return ResponseEntity.ok().body(new AdsDTO(adList.size(), adList));
+    }
+
+    @PatchMapping("/{adId}/image")
```
```suggestion
    @PatchMapping(value = "/{adId}/image", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,152 @@
+    @Operation(summary = "Обновление картинки объявления")
+    public ResponseEntity<Ad> editImageAdById(@PathVariable int adId, @RequestParam String imagePath) {
+        Optional<Ad> ad = adsService.getAdById(adId);
+        if (ad.isEmpty()) return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
```
Это неверно, пустой список - это валидный ответ с кодом 200, а не 404

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,152 @@
+    public ResponseEntity<Ad> editImageAdById(@PathVariable int adId, @RequestParam String imagePath) {
+        Optional<Ad> ad = adsService.getAdById(adId);
+        if (ad.isEmpty()) return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        if (imagePath.isEmpty()) return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
```
никакой логики в контроллерах быть не должно - всё должно быть в сервисах

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,86 @@
+    private CommentsService commentsService;
+
+//    private AdsService adsService;
+    private AdsServiceImpl adsService;
```
```suggestion
    private AdsService adsService;
```
инъекцию надо делать по интерфейсу, а не реализации

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,86 @@
+
+    private MapperUtil mapperUtil;
+
+    public CommentsController(CommentsService commentsService, AdsServiceImpl adsService, MapperUtil mapperUtil) {
```
```suggestion
    public CommentsController(CommentsService commentsService, AdsService adsService, MapperUtil mapperUtil) {
```
инъекцию надо делать по интерфейсу, а не реализации

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,86 @@
+
+    @Operation(summary = "Получение списка всех комментариев")
+    @GetMapping("/all")
+    public ResponseEntity<CommentsDTO> getComment(Ad adsId) {
```
что за параметр? он не нужен

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,86 @@
+
+
+    @Operation(summary = "Получение списка всех комментариев")
+    @GetMapping("/all")
```
такого эндпоинта нет в спецификации

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,86 @@
+    @Operation(summary = "Удаление комментария")
+    @DeleteMapping("/{commentId}")
+//    public void deleteComment(@PathVariable int adId , @RequestParam int commentId ) {
+    public ResponseEntity<?> deleteComment(@PathVariable int adId , @RequestParam int pk ){
```
аннотация последняя неверная, как и название параметра - см. на параметр пути в урле

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,86 @@
+    // обновление комментария
+    @Operation(summary = "Обновление комментария")
+    @PutMapping("/comment/{commentId}")
+    public ResponseEntity<?> updateComment(@PathVariable int adId,@PathVariable int pk, String text) {
```
названия второго параметра неверное - см. на параметр пути в урле. Также комментарий надо передавать в теле запроса, а не в урле

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,86 @@
+
+    // обновление комментария
+    @Operation(summary = "Обновление комментария")
+    @PutMapping("/comment/{commentId}")
```
метод неправильный (см. спецификацию)

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -0,0 +1,91 @@
+
+    @Operation(summary = "Получение информации об авторизованном пользователе")
+    @GetMapping("/me")
+    public ResponseEntity<UserDTO> getUser(@RequestParam String email) {
```
Зачем здесь параметр? он не нужен, см. внимательнее спецификацию, не надо никакую отсебятину делать, иначе ничего работать не будет

Stellareign replied:
++

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/AdsDTO.java:
```diff
@@ -0,0 +1,17 @@
+@NoArgsConstructor
+@AllArgsConstructor
+public class AdsDTO {
+    private int pk;
+    private List<Ad> adList;
```
названия полей неверные (см. спецификацию)

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/comments/CommentDTO.java:
```diff
@@ -0,0 +1,27 @@
+@Data
+@AllArgsConstructor
+public class CommentDTO {
+    private  int count; // общее количество комментариев
```
этого поля нет в DTO (см. спецификацию)

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/user/UserDTO.java:
```diff
@@ -0,0 +1,91 @@
+    private String image;
+    private Role role;
+
+    private String currentPassword;
```
этого поля нет в DTO (см. спецификацию)

Stellareign replied:
++

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/Comments.java:
```diff
@@ -0,0 +1,59 @@
+
+    @OneToOne
+    @JoinColumn (name = "user_id",  nullable = false)
+    private User authorId; // id автора комментария
```
почему у объявления 3 автора?

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/Comments.java:
```diff
@@ -0,0 +1,59 @@
+
+    @ManyToOne
+    @JoinColumn (name = "first_name",  nullable = false)
+    private User authorFirstName;
```
почему у объявления 3 автора?

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/Comments.java:
```diff
@@ -0,0 +1,59 @@
+
+    @ManyToOne
+    @JoinColumn (name = "avatar",  nullable = false)
+    private User authorImage;
```
почему у объявления 3 автора?

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/Image.java:
```diff
@@ -0,0 +1,36 @@
+//    private String imagePath;
+
+    @Column(name = "image_size")
+    private Long imageSize;
```
Эта колонка не нужна - просто длина массива, зачем отдельно хранить?

Stellareign replied:
+

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/User.java:
```diff
@@ -0,0 +1,174 @@
+    private String image; // ссылка на аватар
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "role", nullable = false)
```
role?

dmitry-bizin commented on src/main/java/ru/skypro/homework/exceptions/RecordNotFoundException.java:
```diff
@@ -0,0 +1,10 @@
+package ru.skypro.homework.exceptions;
+
+public class RecordNotFoundException {
```
зачем этот класс? почему он не расширяет ни одно из исключений?

dmitry-bizin commented on src/main/java/ru/skypro/homework/repository/AdsRepository.java:
```diff
@@ -0,0 +1,19 @@
+//    @EntityGraph(attributePaths = {"user"})
+    @EntityGraph(attributePaths = {"author"})
+    @Query(value = "SELECT p FROM Ad p")
+    List<Ad> findAll();
```
зачем изобретать велосипед? метод findAll уже написан в репозиториях, надо просто взять и использовать его

dmitry-bizin commented on src/main/java/ru/skypro/homework/repository/ImageRepository.java:
```diff
@@ -0,0 +1,10 @@
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.stereotype.Service;
+import ru.skypro.homework.entity.Image;
+@EnableJpaRepositories
```
аннотация не нужна

dmitry-bizin commented on src/main/java/ru/skypro/homework/repository/CommentsRepository.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.stereotype.Service;
+import ru.skypro.homework.entity.Comments;
+
+@EnableJpaRepositories
```
аннотация не нужна

dmitry-bizin commented on src/main/java/ru/skypro/homework/repository/CommentsRepository.java:
```diff
@@ -0,0 +1,11 @@
+import ru.skypro.homework.entity.Comments;
+
+@EnableJpaRepositories
+@Service
```
аннотация не нужна

dmitry-bizin commented on src/main/java/ru/skypro/homework/repository/UserRepository.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.stereotype.Service;
+import ru.skypro.homework.entity.User;
+@EnableJpaRepositories
+@Service
```
аннотация не нужна

Stellareign replied:
++

dmitry-bizin commented on src/main/java/ru/skypro/homework/repository/UserRepository.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.stereotype.Service;
+import ru.skypro.homework.entity.User;
+@EnableJpaRepositories
```
аннотация не нужна

Stellareign replied:
++

dmitry-bizin commented on src/main/java/ru/skypro/homework/repository/ImageRepository.java:
```diff
@@ -0,0 +1,10 @@
+import org.springframework.stereotype.Service;
+import ru.skypro.homework.entity.Image;
+@EnableJpaRepositories
+@Service
```
аннотация не нужна

Stellareign replied:
++

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java:
```diff
@@ -0,0 +1,100 @@
+            adsRepository.deleteById(adsId);
+            return true;
+        } else {
+            new RecordNotFoundException(String.valueOf(adsId));
```
что это такое? зачем здесь создаётся объект? какой смысл?

Stellareign committed:
Пытаюсь наладить собщение с БД, поа тольо сохранение пользователя работает, ну и фоток

alkor63 committed:
Merge remote-tracking branch 'origin/Aleksey' into Aleksey

alkor63 committed:
Вношу исправления по замечаниям Наставника

Stellareign committed:
Зменала маппер на фабрики, почти всё заработало.
Как быть с паролем?

Stellareign committed:
Переименовала фабрики

alkor63 committed:
Внёс все исправления по замечаниям Наставника

alkor63 committed:
Merge branch 'Aleksey' into dev

Stellareign committed:
Регистрация и изменение пользователя работает с фронта

Stellareign committed:
Merge branch 'RuAna-2' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/config/MapperUtil.java
#	src/main/java/ru/skypro/homework/controller/AdsController.java
#	src/main/java/ru/skypro/homework/entity/User.java
#	src/main/java/ru/skypro/homework/repository/AdsRepository.java
#	src/main/java/ru/skypro/homework/repository/UserRepository.java

Stellareign committed:
Мерж веток Aleksey и RuAna-2
Изменила getEmail на getUsername

Stellareign committed:
Просто выравнивание сделала

alkor63 committed:
Доработал WebSecurityConfig для работы с БД юзеров

Rumonz1 committed:
исправил код следуя замечаниям наставника

Rumonz1 committed:
поменял аннотацию в entity Comments

Stellareign committed:
Подправила комменты у себя

Stellareign committed:
Несколько правок по адресам в контроллерах

Stellareign committed:
Также (ранее) удалены лишние классы

Stellareign committed:
Заменила колонку author_id на user_id

Stellareign committed:
Merge branch 'RuAna-3' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/entity/Comments.java

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,156 @@
+    }
+
+
+    @PatchMapping("//{id}")
```
2 слэша, должен быть 1

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/comments/CommentsDTO.java:
```diff
@@ -0,0 +1,24 @@
+
+    private int count; // общее количество комментариев
+
+    private Map<Integer, Comments> results; //результат вывода всех найденных комментов
```
здесь неправильно, должен быть список (см. внимательнее спецификацию)

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/AdsDTO.java:
```diff
@@ -0,0 +1,17 @@
+@AllArgsConstructor
+public class AdsDTO {
+    private int count;
+    private List<Ad> results;
```
здесь должно быть dto, а не сущность

alkor63 replied:
Простите, здесь это где?
как мы поняли ТЗ: AdsDTO - это список сущностей Ad и count - количество сущностей в списке.

dmitry-bizin replied:
Здесь - это в списке. Сущности и DTO - абсолютно разные вещи и в этой домашке про сущности речь вообще не шла, это тема второго занятия. В этой работе надо создать DTO-классы согласно спецификации. В этом же классе используется сущность, что неправильно, потому что сущности и DTO не должны мешаться, иначе смысл в DTO теряется

alkor63 replied:
Дима, привет! 
А какую ДТО нужно показывать фронту (возвращать в контроллер) при создании и/или обновлении объявления?
CreateOrUpdateAdDTO где всего 3 поля ?

dmitry-bizin replied:
Нет, нужная DTO просто называется Ad в спецификации

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/user/UpdatePasswordDTO.java:
```diff
@@ -0,0 +1,25 @@
+@NoArgsConstructor
+public class UpdatePasswordDTO {
+    /*
+в сваггере тип stringst - это что?
```
там строка

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/Comments.java:
```diff
@@ -0,0 +1,67 @@
+    private int pk;    // id комментания
+
+    @Column(name = "author_first_name", nullable = false)
+    private String authorFirstName;
```
дублирование данных - это уже есть в юзере

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/Comments.java:
```diff
@@ -0,0 +1,67 @@
+    private String authorFirstName;
+
+    @Column(name = "author_avatar", nullable = false)
+    private String authorImage; // ссылка н аватар автора
```
дублирование данных - это уже есть в юзере

dmitry-bizin commented on src/main/java/ru/skypro/homework/exceptions/RecordNotFoundException.java:
```diff
@@ -0,0 +1,6 @@
+package ru.skypro.homework.exceptions;
+
+public class RecordNotFoundException extends Throwable {
```
```suggestion
public class RecordNotFoundException extends RuntimeException {
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/Comments.java:
```diff
@@ -0,0 +1,67 @@
+@Entity
+@Getter
+@Table(name = "comments")
+public class Comments {
```
```suggestion
public class Comment {
```
названия классов должны быть в единственном числе

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/MyUserDetailes.java:
```diff
@@ -0,0 +1,51 @@
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return null;
```
здесь надо вернуть authority юзера

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/MyUserDetailes.java:
```diff
@@ -0,0 +1,51 @@
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return false;
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return false;
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return false;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return false;
```
должно быть true везде

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java:
```diff
@@ -0,0 +1,93 @@
+
+    @Override
+    public Ad addAd(Ad ad) {
+        adsRepository.save(ad);
+        return ad;
+    }
+
+    @Override
+    public Ad editAdById(int id, CreateOrUpdateAdDTO updateAd){
+        Optional<Ad> optionalAd = adsRepository.findById(id);
+        if (optionalAd.isEmpty()) {
+            new RecordNotFoundException(String.valueOf(id));
+        }
+        Ad existingAd = optionalAd.get();
+        existingAd.setTitle(updateAd.getTitle());
+        existingAd.setPrice(updateAd.getPrice());
+        existingAd.setDescription(updateAd.getDescription());
+        adsRepository.save(existingAd);
+        return existingAd;
+    }
+
+    @Override
+    public Ad editImageAdById(int id, String image) {
+        Optional<Ad> optionalAd = adsRepository.findById(id);
+        if (optionalAd.isEmpty()) {
+            new RecordNotFoundException(String.valueOf(id));
+        }
+        Ad existingAd = optionalAd.get();
+        existingAd.setImage(image);
+        adsRepository.save(existingAd);
+        return existingAd;
+    }
+
+    @Override
+    public List<Ad> getAllAdsByUser(String userLogin) {
+        return adsRepository.findAll()
+                .stream()
+                .filter(e -> e.getAuthor().getUsername().equals(userLogin))
+                .collect(Collectors.toList());
+    }
```
методы сервисов должны принимать и возвращать DTO, вся логика должна быть внутри методов сервисов, сущности наружу не показываются, только DTO

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AuthServiceImpl.java:
```diff
@@ -1,23 +1,31 @@
 
     private final UserDetailsManager manager;
     private final PasswordEncoder encoder;
+    public static boolean AUTHORISE = false;
```
что это такое?

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/CommentsServiceImpl.java:
```diff
@@ -0,0 +1,76 @@
+        this.commentsRepository = commentsRepository;
+    }
+
+    private final Map<Integer, Comments> commentsMap = new HashMap<>();
```
что это такое?

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/CommentsServiceImpl.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    @Override
+    public Comments getComment(int pk) {
+        return commentsRepository.findById(pk).orElseThrow();
+    }
+
+    @Override
+    public Comments addComment(String text, Ad adId) {
+        Comments comments = new Comments(text, adId);
+        return commentsRepository.save(comments);
+    }
+
+    @Override
+    public boolean deleteComment(int pk) {
+        Optional<Comments> optionalComments = commentsRepository.findById(pk);
+        if (optionalComments.isPresent()) {
+            commentsRepository.deleteById(pk);
+            return true;
+        } else {
+            new RecordNotFoundException(String.valueOf(pk));
+            return false;
+        }
+
+    }
+
+    @Override
+    public Comments updateComment(int pk, CreateOrUpdateCommentDTO createOrUpdateCommentDTO) {
+        Optional<Comments> commentsOptional = commentsRepository.findById(pk);
+        if (commentsOptional.isEmpty()){
+            new RecordNotFoundException(String.valueOf(pk));
+        }
+        Comments existingComm = commentsOptional.get();
+        existingComm.setText(createOrUpdateCommentDTO.getText());
+        return commentsRepository.save(existingComm);
```
методы сервисов должны принимать и возвращать DTO, всяк логика должна быть внутри методов сервисов, сущности наружу не показываются, только DTO

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    @Override
+    public User getUserByUsernameFromDB(String username) {
+        return userRepository.findByUsername(username);
+    }
+
+
+    @Override
+    public User updateUser(User user, UpdateUserDTO updateUserDTO) {
+        user.setFirstName(updateUserDTO.getFirstName());
+        user.setLastName(updateUserDTO.getLastName());
+        user.setPhone(updateUserDTO.getPhone());
+        userRepository.save(user);
+        return user;
+    }
+
+    @Override
+    public void saveRegisterUser(Register register) {
+        userRepository.save(createUserFromRegister(register));
+    }
+
+    private User createUserFromRegister(Register register) {
+        String pass = encoder.encode(register.getPassword()); // шифрование пароля
+        User user = new User();
+        user.setUsername(register.getUsername());
+        user.setFirstName(register.getFirstName());
+        user.setLastName(register.getLastName());
+        user.setRole(register.getRole());
+        user.setPhone(register.getPhone());
+        user.setPassword(pass);
+        user.setRegisterDate(LocalDate.from(LocalDate.now()));
+        return user;
```
методы сервисов должны принимать и возвращать DTO, вся логика должна быть внутри методов сервисов, сущности наружу не показываются, только DTO

Stellareign replied:
Дима, привет. 
Так это метод приватный и используется в void методе. Всё равно он должен возвращать dto?

dmitry-bizin replied:
Не, тогда всё норм, проглядел)

Stellareign replied:
Ок))

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/CommentsServiceImpl.java:
```diff
@@ -0,0 +1,76 @@
+import java.util.stream.Collectors;
+
+@Service
+public class CommentsServiceImpl implements CommentsService {
```
```suggestion
public class CommentServiceImpl implements CommentsService {
```

dmitry-bizin commented on src/main/resources/hibernate.cfg.xml:
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE hibernate-configuration PUBLIC
```
этот файл не нужен

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,17 @@
+spring.datasource.url=jdbc:postgresql://localhost:5432/diplom
+spring.datasource.username=postgres
+spring.datasource.password=1234
+spring.jpa.hibernate.ddl-auto=update
```
при использовании ликвибейз стратегия должна быть validate

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,17 @@
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
+
+#pereopredeenie binov:
+spring.main.allow-bean-definition-overriding = true
```
это надо удалить и писать бины, чтобы они были уникальны по названиями

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,17 @@
+#pereopredeenie binov:
+spring.main.allow-bean-definition-overriding = true
+
+#vyvodit telo zaprosa v konsol'
```
не тело запроса, а sql

alkor63 commented:
То есть Ad это и сущность и ДТО?или сущность для объявлений какая-то третья? 13.10.2023, 14:02, "Dmitry Bizin" ***@***.***>***@***.*** commented on this pull request.In src/main/java/ru/skypro/homework/dto/ads/AdsDTO.java:> +package ru.skypro.homework.dto.ads;
+
+import lombok.*;
+import ru.skypro.homework.entity.Ad;
+
+import java.util.List;
+
+/*
+За Алексеем
+ */
***@***.***
***@***.***
***@***.***
+public class AdsDTO {
+    private int count;
+    private List<Ad> results;
Нет, нужная DTO просто называется Ad в спецификации—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

BizinMitya commented:
Нет. Есть сущность, а есть DTO. В контроллерах должны использоваться ТОЛЬКО DTO.

alkor63 commented:
Про ДТО в контроллерах я понял.В спецификации нам чётко прописаны некие классы. Что из них сущность Объявление, а что ДТО Объявления, например,  при создании нового объявления?14:59, 13 октября 2023 г., Dmitry Bizin ***@***.***>:
Нет. Есть сущность, а есть DTO. В контроллерах должны использоваться ТОЛЬКО DTO.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
-- Отправлено из мобильного приложения Яндекс Почты

BizinMitya commented:
В спецификации прописаны только DTO, сущности надо продумать самостоятельно исходя из модели данных, представленной DTO. Ещё раз повторю - неверно создавать сущности один в один как DTO.

Stellareign committed:
Правки по замечаниям наставника.
+ Что-то сделала по аутенфикации, приложение компилируетсяя, юзер сохраняется, можно менять пароль, но в ЛК входит по любму паролю...

Stellareign committed:
Изменила в методе updateUser ходной параметр User на String username
Добавила входной параметр Authentication authentication в контроллеры, изменила код в контроллерах под authentication
Добавила метод getUserForGetController(String username)
Изменила доступ к методу User getUserByUsernameFromDB
Кое-где в методах добавила логи

alkor63 committed:
Добавил класс AdEntity, теперь Ad - это DTO
ВСЮ логику из контроллера убрал в сервис

Stellareign committed:
Авторизация / аутентификация работает!
Пользователь обновляется, пароль меняется с проверкой соответствия условий, с неправильным паром в ЛК не пускает! :)

Stellareign committed:
Добавила сущность AdsImage
Переименовала Image в Avatar
Пока пробую AdsImage и Avatar сохранить в одной таблице images

Stellareign committed:
Написала немного JavaDock по сервису юзера
Заменила ссылку на аватар на сам аватар (возможно, придётся вернуть обратно)
Начала обрабатывать картинку

alkor63 committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/controller/AdsController.java
#	src/main/java/ru/skypro/homework/entity/AdEntity.java
#	src/main/java/ru/skypro/homework/exceptions/RecordNotFoundException.java
#	src/main/java/ru/skypro/homework/repository/AdsRepository.java

Stellareign committed:
Удалила дублированный класс картинки

Stellareign committed:
Исправила фабрики юзера

Stellareign committed:
Вернула ссылку на аватар.
У юзера аватар подцепился в БД.
Во фронте не обновился.
В таблице картинок не появился.

Rumonz1 committed:
6:17 утра по мск, закончил работу над ошибками и чуть продвинулся.

Rumonz1 committed:
6:17 утра по мск, закончил работу над ошибками и чуть продвинулся.

Stellareign committed:
Merge branch 'Timur2.0revork' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/repository/AdsRepository.java

Stellareign committed:
Пытаюсь что-то сделать с аватаром. В БД сохраняется, на фронт - нет

Stellareign committed:
Пытаюсь что-то сделать с аватаром. В БД сохраняется

Stellareign committed:
Пытаюсь что-то сделать с аватаром. В БД сохраняется

Stellareign committed:
Merge branch 'RuAna-3' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/config/WebSecurityConfig.java
#	src/main/java/ru/skypro/homework/controller/AdsController.java
#	src/main/java/ru/skypro/homework/controller/CommentsController.java
#	src/main/java/ru/skypro/homework/dto/comments/CommentsDTO.java
#	src/main/java/ru/skypro/homework/entity/AdEntity.java
#	src/main/java/ru/skypro/homework/entity/AdsImage.java
#	src/main/java/ru/skypro/homework/entity/Comment.java
#	src/main/java/ru/skypro/homework/repository/CommentsRepository.java
#	src/main/java/ru/skypro/homework/service/impl/CommentsServiceImpl.java
#	src/main/java/ru/skypro/homework/service/interfaces/CommentsService.java

Stellareign committed:
Удалила изображения

alkor63 committed:
Добавил аутентификацию в AdsController

alkor63 committed:
Добавил аутентификацию в AdsController
проверил коды ответов

alkor63 committed:
Добавил аутентификацию в AdsController
проверил коды ответов

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/MapperUtil.java:
```diff
@@ -1,19 +1,32 @@
+@NoArgsConstructor
 @Configuration
 public class MapperUtil {
+    private AdsImageService adsImageService;
```
зачем в маппере сервис? это неправильно, в мапперах могут быть только другие мапперы, но не сервисы

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -34,123 +30,72 @@ public class AdsController {
-          else {
-                return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
-            }
+    public ResponseEntity<?> createAd(@RequestBody String title,   // 'заголовок объявления'
+                                      int price,               // 'цена объявления'
+                                      String description,            //'описание объявления'
+                                      MultipartFile image) throws IOException             //'адрес картинки объявления'
+    //                                      User author)          //'id автора объявления'
```
по-прежнему неправильно

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -32,55 +32,66 @@ public class CommentsController {
+            CommentsDTO commentsDTO = commentsService.getAllComments(adId);
+            return ResponseEntity.ok(commentsDTO);
+        } catch (RuntimeException e) {
+            e.getStackTrace();
```
зачем эта строчка? что она делает?

Rumonz1 replied:
Я правильно понимаю, что претензия именно к строке catch ?

dmitry-bizin replied:
к первой строке внутри catch, да

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -32,55 +32,66 @@ public class CommentsController {
+            CommentDTO commentDTO = commentsService.addComment(COUComment, pk, userInfo);
+            return ResponseEntity.ok(commentDTO);
+        } catch (RuntimeException e) {
+            e.getStackTrace();
```
зачем эта строчка? что она делает?

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -32,55 +32,66 @@ public class CommentsController {
+            commentsService.deleteComment(adId, pk);
+            return ResponseEntity.ok().build();
+        } catch (RuntimeException e) {
+            e.getStackTrace();
```
зачем эта строчка? что она делает?

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -32,55 +32,66 @@ public class CommentsController {
+              commentsService.updateComment(adId, pk, COUComment);
+              return ResponseEntity.ok().build();
+          } catch (RuntimeException e) {
+              e.getStackTrace();
```
зачем эта строчка? что она делает?

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -32,55 +32,66 @@ public class CommentsController {
-        return ResponseEntity.ok(commentsService.updateComment(pk, updateComment));
+        @Operation(summary = "Обновление комментария")
+        @PatchMapping("/{adId}/comments/{commentId}")
+    public ResponseEntity<CommentDTO> updateComment(@PathVariable int adId,@PathVariable int pk,
```
pk - неправильное название параметра, такого параметра нет в пути

Rumonz1 replied:
Понял, когда писал, думал поменять на commentId , но решил так оставить, получается всё же исправить надо было?

dmitry-bizin replied:
маппинг происходит либо по названию переменной, либо через аннотацию. Т.е.
```suggestion
    public ResponseEntity<CommentDTO> updateComment(@PathVariable int adId, @PathVariable int commentId,
```
или
```suggestion
    public ResponseEntity<CommentDTO> updateComment(@PathVariable int adId, @PathVariable("commentId") int pk,
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -32,55 +32,66 @@ public class CommentsController {
-            return new ResponseEntity<>(("Комментарий с id = " + pk + "успешно удалён"), HttpStatus.NO_CONTENT);
-        } else {
-            return new ResponseEntity<>(("Ошибка при попытке удалить комментарий с id = " + pk), HttpStatus.NOT_FOUND);
+    public ResponseEntity<CommentDTO> deleteComment(@PathVariable int adId , @PathVariable int pk ) {
```
pk - неправильное название параметра, такого параметра нет в пути

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -32,55 +32,66 @@ public class CommentsController {
-        if (pk != null) {
-            return ResponseEntity.ok(pk);
-        } else return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+    public ResponseEntity<CommentDTO> addComment(@PathVariable Integer pk,
```
pk - неправильное название параметра, такого параметра нет в пути

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -32,55 +32,66 @@ public class CommentsController {
-            return ResponseEntity.ok().body(commentsDTO);
-        } else return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+    @GetMapping("/{id}/comments")
+    public ResponseEntity<CommentsDTO>  getComments(@PathVariable int adId) {
```
adId - неправильное название параметра, такого параметра нет в пути

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/Ad.java:
```diff
@@ -0,0 +1,14 @@
+    private int price;                      // 'цена объявления'
+    private String title;                   // 'заголовок объявления'
+    private User author;          // автор объявления
+    private AdsImage image;                   //'ссылка на картинку объявления'
```
Это неверно, в спецификации нет такого поля

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/Ad.java:
```diff
@@ -0,0 +1,14 @@
+    private int pk;                         //'id объявления'
+    private int price;                      // 'цена объявления'
+    private String title;                   // 'заголовок объявления'
+    private User author;          // автор объявления
```
Это неверно, в спецификации нет такого поля

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/AdsDTO.java:
```diff
@@ -13,5 +13,5 @@
 public class AdsDTO {
     private int count;
-    private List<Ad> results;
+    private List<AdEntity> results;
```
в dto не должно быть никаких сущностей, писал про это в прошлый раз

alkor63 replied:
Дима, а как правильно понимать эту схему:
Ads  {
     count              integer
                            общее количество объявлений
                            
      results              [Ad > {...}]
      }                      
Что здесь Ad ?
Поля все такие же как у Ad (DTO)

dmitry-bizin replied:
Там ссылка же есть в спецификации, это DTO Ad

dmitry-bizin replied:
![image](https://github.com/Stellareign/DIPLOMA_WORK/assets/15345076/96127771-e5d4-4552-8029-e2eff7b4019c)
вот ссылка, надо просто кликнуть и перекинет

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/ExtendedAdDTO.java:
```diff
@@ -10,7 +12,7 @@ public class ExtendedAdDTO {
     private int price;
     private  String title;
-    private String image; // фото товара
+    private AdsImage image; // фото товара
```
Это неверно, в спецификации нет такого поля

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/comments/CommentDTO.java:
```diff
@@ -4,25 +4,27 @@
     private String authorFirstName;//    имя создателя комментария
 
-    private long createdAt; // дата и время создания комментария в миллисекундах с 00:00:00 01.01.1970
+    private LocalDateTime createdAt; // дата и время создания комментария в миллисекундах с 00:00:00 01.01.1970
```
тип поля неверный (см. спецификацию)

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/Avatar.java:
```diff
@@ -14,31 +14,35 @@
-//    private String imagePath;
-
+    @Column(name = "image_type")
+    private String fileType;
```
это поле не нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/User.java:
```diff
@@ -52,12 +49,17 @@ public class User {
+//    private Avatar userAvatar;
+
+    @Column(name = "avatar_path")
+    String avatarPath;  //
```
все поля должны быть приватными

dmitry-bizin commented on src/main/java/ru/skypro/homework/repository/AdsRepository.java:
```diff
@@ -1,10 +1,13 @@
-
-public interface AdsRepository extends JpaRepository<Ad, Integer> {
+@EnableJpaRepositories
+@Service
```
как репозиторий может быть сервисом?

dmitry-bizin commented on src/main/java/ru/skypro/homework/security_service/MyUserDetails.java:
```diff
@@ -0,0 +1,61 @@
+
+@Component
+public  class MyUserDetails implements UserDetails {
+    private final User user;
```
сущность здесь использовать неверно, лучше завести dto для этого

Stellareign replied:
Дима, я правильно понимаю, что лучше создать дополнительную dto, например, с полями username, password  и role, или  нужно / можно использовать существующую UserDTO?

dmitry-bizin replied:
Дополнительную, потому что она нужна будет для использования здесь, а та нужна для использования на фронте

Stellareign replied:
Ну я вчера так и сделала )

dmitry-bizin commented on src/main/java/ru/skypro/homework/security_service/MyUserDetails.java:
```diff
@@ -0,0 +1,61 @@
+ */
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return Set.of(new SimpleGrantedAuthority(user.getRole().name()));
```
authorities должны начинаться с префикса ROLE_

dmitry-bizin commented on src/main/java/ru/skypro/homework/security_service/SecurityUserDetailsService.java:
```diff
@@ -0,0 +1,32 @@
+@Service
+public class SecurityUserDetailsService implements UserDetailsService {
+
+    @Autowired
```
инъекцию лучше делать через конструктор

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdsImageServiceImpl.java:
```diff
@@ -0,0 +1,78 @@
+    public AdsImage getImage(int id) throws RecordNotFoundException {
+        log.info("Метод извлекает изображение объявления с id = {}", id);
+        Optional<AdsImage> adsImage = adsImageRepository.findById(id);
+if (adsImage.isEmpty()) throw new RecordNotFoundException("Не удалось найти картинку товара с id =  "+id);
```
фигурные скобки надо всегда писать

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java:
```diff
@@ -1,83 +1,143 @@
-    public Optional<Ad> getAdById(int adsId) {
-        return adsRepository.findById(adsId);
+    public ExtendedAdDTO getAdById(int adsId) {
+        return mapperUtil.createExtendedAdDTO(adsRepository.findById(adsId).get());
```
неправильная работа с Optional - будет исключение, если объекта внутри не будет

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/MapperUtil/CommentMapping.java:
```diff
@@ -0,0 +1,38 @@
+
+    }
+
+    public Comment mapToEntity(CreateOrUpdateCommentDTO COUComment, User author, AdEntity ad) {
```
названия переменных надо писать с маленькой буквы

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java:
```diff
@@ -78,5 +177,58 @@ private User createUserFromRegister(Register register) {
+        if (username.matches("^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$")) {
+            log.info("Логин указан верно");
+            return true;
+        } else log.info("Проверьте указанный email. Логин должен быть указан в формате user@user.us!");
```
фигурные скобки надо всегда писать

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java:
```diff
@@ -78,5 +177,58 @@ private User createUserFromRegister(Register register) {
+        if (phone.matches("\\+7\\s?\\(\\d{3}\\)\\s?\\d{3}-\\d{2}-\\d{2}")) {
+            log.info("Формат телефона верный.");
+            return true;
+        } else log.info("Укажите номер телефона в формате +7(ХХХ)ХХХ-ХХ-ХХ!");
```
фигурные скобки надо всегда писать

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java:
```diff
@@ -78,5 +177,58 @@ private User createUserFromRegister(Register register) {
+        if (password.length() >= 8 && !password.isBlank()) {
+            log.info("Пароль соответствует требованиям!");
+            return true;
+        } else log.info("Пароль не соответствует требованиям! Пароль не должен состоять из пробелов, длина" +
```
фигурные скобки надо всегда писать

alkor63 committed:
Внёс исправления по замечаниям наставника

alkor63 committed:
Merge branch 'Alexey4' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/repository/CommentRepository.java
